KerbalStuff/blueprints/admin.py

```diff
@@ -19,7 +24,55 @@
 @admin.route("/admin")
 @adminrequired
 def admin_main() -> werkzeug.wrappers.Response:
-    return redirect(url_for('admin.users', page=1))
+    return redirect(url_for('admin.profiling', page=1))
+
+
+@admin.route("/admin/profiling/<int:page>")
+@adminrequired
+def profiling(page: int) -> Union[str, werkzeug.wrappers.Response]:
+    # TODO: query: min duration, time window, route patterns
+    if page < 1:
+        return redirect(url_for('admin.profiling', page=1, **request.args))
+    prof_dir = _cfg('profile-dir')
+    profilings = [] if not prof_dir else list(map(
+        parse_prof_filename,
+        sorted(Path(prof_dir).glob('*.prof'),
+               key=lambda p: -p.stat().st_mtime)))
+    total_pages = max(1, math.ceil(len(profilings) / ITEMS_PER_PAGE))
+    profilings = profilings[(page - 1) * ITEMS_PER_PAGE : page * ITEMS_PER_PAGE]
+    return render_template("admin-profiling.html", profilings=profilings, page=page, total_pages=total_pages)
+
+
+def parse_prof_filename(p: Path) -> Dict[str, Any]:
+    pieces = p.name.split('.')
+    route_pieces = pieces[1:-3]
+    # ProfilerMiddleware uses 'root' as the route when it's '/', which doesn't help us
+    if len(route_pieces) == 1 and route_pieces[0] == 'root':
+        route_pieces = []
+    return {
+        'name': p.name,
+        'route': '/' + '/'.join(route_pieces),
```

Comment:
Minor issue with this, routes that actually have a dot in their name, like images under `/content`, also have it replaced with a slash:
```
/content/user_2/Some_mod/Some_mod-1615741419/4325244/png
/content/admin_1/admins_test_mod/thumb_admins_test_mod-1605370308/307536/jpg
/content/user_2/Some_mod/thumb_Some_mod-1615741419/4325244/jpg
```
We could adjust the `filename_format` to use something other that `.` as separator.

Alternatively, we could exclude `/content` from profiling as well, since those are all static files.
Maybe we should do that anyway, because on alpha+beta+production downloads redirect to a `/content` path as well, which we don't really care about.

Reply:
Excluding `/content` for now. We can do something more complicated if we find we want to profile that path later.